### PR TITLE
Add middleware and error handling

### PR DESCRIPTION
**Pull request by vderunov**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -10,16 +10,19 @@ const DATA_DIR = path.join(__dirname, 'data');
...
-    res.status(400).send({ message: 'Invalid wallet address' });
-    return false;
+    return next({ code: 400, message: 'Invalid wallet address' });
   }
-  return true;
+  next();
+};
+
+const transformWalletAddress = (req, res, next) => {
+  req.body.walletAddress = `${req.body.walletAddress.toLowerCase()}.unverified`;
```
Comment (noisekit) [inline server.js:24]:
```suggestion
  req.body.walletAddress = `${req.body.walletAddress.toLowerCase()}`;
```

File: server.js [hunk 2]
```diff
@@ -10,16 +10,19 @@ const DATA_DIR = path.join(__dirname, 'data');
 
 app.use(express.json());
 
-const validateWalletAddress = (req, res) => {
+const validateWalletAddress = (req, res, next) => {
   if (!req.body.walletAddress) {
-    res.status(400).send({ message: 'Missing wallet address' });
-    return false;
+    return next({ code: 400, message: 'Missing wallet address' });
```
Comment (noisekit) [inline server.js:15]:
```suggestion
    return next(Object.assign(new Error('Missing wallet address', { code: 400 })));
```
Always use real error instances. But feel free to extend them with extra properties.

**Review by noisekit — APPROVED** [review]

File: server.js (hunk 2 above)
Comment (vderunov) [reply server.js]: Added new class:

```js
class HttpError extends Error {
  constructor(message, code) {
    super(message);
    this.code = code;
  }
}
```

usage:

```js
  return next(new HttpError('Missing wallet address', 400));
```

File: server.js (hunk 1 above)
Comment (vderunov) [reply server.js]: fixed

**Review by noisekit — APPROVED** [review]

**Pull request merged**